transport/grpc/config.go

```diff
@@ -95,6 +131,8 @@ type InboundConfig struct {
 //  outbounds:
 //    myservice:
 //      grpc:
+//        tls:
```

Comment:
How would I configure outbound mTLS?

Reply:
I just pushed additional documentation/examples in doc.go - the API is richer than this config supports in this PR.

Reply:
@bradbl As @zmt points out, the config structure we’re proposing in this change is not as expressive as the underlying API, which allows you to thread an arbitrary credentials object into a peer list.

You could give us a big hand by starting a follow-up discussion about what configuration you expect to be necessary to use mTLS, if any. It might alternately be possible to thread a dependency, like a credentials provider, that we can put in an application’s dependency container and automatically set up credentials based on the outbound service name. Please let us know what you’ll need.

Reply:
Do you expect that the configuration will be forwards compatible with including those options someday?  I don't see any reason why it wouldn't be from what I've read - just want to make sure. For example, someday supporting:

```yaml
outbounds:
  myservice:
    grpc:
      tls:
        enabled:true
        certFile: "/path/to/cert"
        keyFile: "/path/to/key"
```

Reply:
For Uber's purposes, injecting credentials via API is almost certainly necessary, as we need to do things like renew the certificate in the background, and supply custom peer verification logic.

Reply:
We have the option of adding fields to the config for static cases. I expect that the latter idea of having a credentials provider will be necessary. Will it be sufficient to thread a credentials object. I’m hoping that the credentials.TransportCredentials interface gives you enough flexibility to rotate credentials behind the scenes.

I would not be surprised but mildly disappointed if the RPC layer needs to be aware of the credential expiration times so it can anticipate expiration, drain stale connections, and create new ones when we receive credential updates. That would entail a great deal more API surface.

Reply:
You raise a good question. If the certificate is short-lived and we need to reload with a fresher certificate - does the API currently allow this? Same question for inbound, except the inbound can be configured via both the API and YAML.

Reply:
Naively, I expect the the certificates could be refreshed by using a credentials.TransportCredentials created from a tls.Config instance which wires up appropriate functions for GetCertificate and GetClientCertificate. I could be mistaken though. A unit/integration test of this would be good.

Reply:
Updated the integration test to use GetCertificate and GetClientCertificate callbacks. Confirmed working.